crates/analysis/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+use sightglass_data::{Measurement, Phase, Summary};
+use std::{borrow::Cow, collections::HashSet};
+
+/// Summarize measurements grouped by: architecture, engine, benchmark file, phase and event.
+pub fn aggregate<'a>(measurements: &[Measurement<'a>]) -> Vec<Summary<'a>> {
+    let mut summaries = Vec::new();
+    for (arch, engine, wasm, phase, event) in keys(&measurements) {
+        let grouped_counts: Vec<_> = measurements
+            .iter()
+            .filter(|m| m.phase == phase && m.event == event)
```

Comment:
I think this needs to filter `m.arch == arch` and similar for `engine` and `wasm` as well?

Reply:
Oh, right; let me update the test to catch that.